0.3.140.3618-203-nightly / Intellij IDEA 2020.3.1   file delete via IDE errors

<!--
Hello and thank you for the issue!
If you would like to report a bug, we have added some points below that you can fill out.
Consider using "Help | Create New Issue" action that fills out all necessary information automatically.
Feel free to remove all the irrelevant text to request a new feature.
-->

## Environment

* IntelliJ Rust plugin version: 0.3.140.3618-203-nightly
* Rust toolchain version: 1.48.0
* IDE name and version: IntelliJ IDEA 2020.3.1  Build #IU-203.6682.168
* Operating system: MacOS 10.15.7

## Problem description
Hey this plug-in is awesome! Anyway, after updating the Rust plug-in I'm no longer able to delete files. Here's the relevant bits of my idea log.

> 2021-01-20 10:02:03,405 [  13361]  ERROR - ellij.ide.actions.DeleteAction - org/toml/lang/psi/TomlKeySegment 
java.lang.NoClassDefFoundError: org/toml/lang/psi/TomlKeySegment
	at org.rust.toml.CargoTomlElementDescriptionProvider.getElementDescription(CargoTomlElementDescriptionProvider.kt:20)
	at com.intellij.psi.ElementDescriptionUtil.getElementDescription(ElementDescriptionUtil.java:17)
	at com.intellij.ide.util.DeleteUtil.generateWarningMessage(DeleteUtil.java:27)
	at com.intellij.refactoring.safeDelete.SafeDeleteDialog.createNorthPanel(SafeDeleteDialog.java:74)
	at com.intellij.openapi.ui.DialogWrapper.init(DialogWrapper.java:1326)
	at com.intellij.refactoring.safeDelete.SafeDeleteDialog.<init>(SafeDeleteDialog.java:49)
	at com.intellij.ide.util.DeleteHandler$2.<init>(DeleteHandler.java:133)
	at com.intellij.ide.util.DeleteHandler.deletePsiElement(DeleteHandler.java:121)
	at com.intellij.ide.util.DeleteHandler.deletePsiElement(DeleteHandler.java:108)
	at com.intellij.ide.projectView.impl.ProjectViewDeleteElementProvider.deleteElement(ProjectViewDeleteElementProvider.java:53)
	at com.intellij.ide.actions.DeleteAction.actionPerformed(DeleteAction.java:37)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:281)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:310)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:286)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:77)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:299)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:110)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:95)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:110)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:514)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:544)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6652)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6417)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:911)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:840)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:507)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.ClassNotFoundException: org.toml.lang.psi.TomlKeySegment PluginClassLoader[PluginDescriptor(name=Rust, id=org.rust.lang, path=/Users/m0n0096/Library/Application Support/JetBrains/IntelliJIdea2020.3/plugins/intellij-rust, version=0.3.140.3618-203-nightly)] com.intellij.ide.plugins.cl.PluginClassLoader@6629530
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:129)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 60 more
2021-01-20 10:02:03,406 [  13362]  ERROR - ellij.ide.actions.DeleteAction - IntelliJ IDEA 2020.3.1  Build #IU-203.6682.168 
2021-01-20 10:02:03,406 [  13362]  ERROR - ellij.ide.actions.DeleteAction - JDK: 11.0.9.1; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2021-01-20 10:02:03,406 [  13362]  ERROR - ellij.ide.actions.DeleteAction - OS: Mac OS X 
2021-01-20 10:02:03,406 [  13362]  ERROR - ellij.ide.actions.DeleteAction - Plugin to blame: Rust version: 0.3.140.3618-203-nightly 
2021-01-20 10:02:03,406 [  13362]  ERROR - ellij.ide.actions.DeleteAction - Last Action: $Delete 

## Steps to reproduce

1. Install 0.3.140.3618-203-nightly , restart idea
2. try to delete any file via the ide
3. file a silly github issue

--- a/toml/src/main/kotlin/org/rust/toml/CargoTomlElementDescriptionProvider.kt
+++ b/toml/src/main/kotlin/org/rust/toml/CargoTomlElementDescriptionProvider.kt
@@ -17,6 +17,7 @@ import org.toml.lang.psi.TomlKeySegment
 
 class CargoTomlElementDescriptionProvider : ElementDescriptionProvider {
     override fun getElementDescription(element: PsiElement, location: ElementDescriptionLocation): String? {
+        if (!tomlPluginIsAbiCompatible()) return null
         return if (element is TomlKeySegment) {
             if (element.isFeatureDef) {
                 when (location) {

--- a/toml/src/main/kotlin/org/rust/toml/inspections/CargoTomlCyclicFeatureInspection.kt
+++ b/toml/src/main/kotlin/org/rust/toml/inspections/CargoTomlCyclicFeatureInspection.kt
@@ -10,6 +10,7 @@ import com.intellij.codeInspection.ProblemsHolder
 import com.intellij.psi.PsiElementVisitor
 import org.rust.cargo.CargoConstants
 import org.rust.toml.isFeatureListHeader
+import org.rust.toml.tomlPluginIsAbiCompatible
 import org.toml.lang.psi.*
 import org.toml.lang.psi.ext.TomlLiteralKind
 import org.toml.lang.psi.ext.kind
@@ -23,8 +24,10 @@ import org.toml.lang.psi.ext.kind
  * ```
  */
 class CargoTomlCyclicFeatureInspection : LocalInspectionTool() {
-    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
-        buildVisitor(holder) ?: super.buildVisitor(holder, isOnTheFly)
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        if (!tomlPluginIsAbiCompatible()) return super.buildVisitor(holder, isOnTheFly)
+        return buildVisitor(holder) ?: super.buildVisitor(holder, isOnTheFly)
+    }
 
     private fun buildVisitor(holder: ProblemsHolder): PsiElementVisitor? {
         if (holder.file.name != CargoConstants.MANIFEST_FILE) return null

--- a/toml/src/main/kotlin/org/rust/toml/search/CargoTomlFindUsagesHandlerFactory.kt
+++ b/toml/src/main/kotlin/org/rust/toml/search/CargoTomlFindUsagesHandlerFactory.kt
@@ -9,10 +9,12 @@ import com.intellij.find.findUsages.FindUsagesHandler
 import com.intellij.find.findUsages.FindUsagesHandlerFactory
 import com.intellij.psi.PsiElement
 import org.rust.toml.isFeatureDef
+import org.rust.toml.tomlPluginIsAbiCompatible
 import org.toml.lang.psi.TomlKeySegment
 
 class CargoTomlFindUsagesHandlerFactory : FindUsagesHandlerFactory() {
     override fun canFindUsages(element: PsiElement): Boolean {
+        if (!tomlPluginIsAbiCompatible()) return false
         return element is TomlKeySegment && element.isFeatureDef
     }
 

--- a/toml/src/main/kotlin/org/rust/toml/search/CargoTomlUsageTypeProvider.kt
+++ b/toml/src/main/kotlin/org/rust/toml/search/CargoTomlUsageTypeProvider.kt
@@ -12,12 +12,14 @@ import com.intellij.usages.impl.rules.UsageTypeProviderEx
 import org.rust.lang.core.RsPsiPattern.anyCfgFeature
 import org.rust.lang.core.psi.RsLitExpr
 import org.rust.toml.CargoTomlPsiPattern.onFeatureDependencyLiteral
+import org.rust.toml.tomlPluginIsAbiCompatible
 import org.toml.lang.psi.TomlLiteral
 
 class CargoTomlUsageTypeProvider : UsageTypeProviderEx {
     override fun getUsageType(element: PsiElement?): UsageType? = getUsageType(element, UsageTarget.EMPTY_ARRAY)
 
     override fun getUsageType(element: PsiElement?, targets: Array<out UsageTarget>): UsageType? {
+        if (!tomlPluginIsAbiCompatible()) return null
         return when (element) {
             is TomlLiteral -> if (onFeatureDependencyLiteral.accepts(element)) FEATURE_DEPENDENCY else DEPENDENCY_FEATURE
             is RsLitExpr -> if (anyCfgFeature.accepts(element)) CFG_FEATURE else null

